source/server/factory_context_impl.cc

```diff
@@ -6,13 +6,9 @@ namespace Server {
 FactoryContextImplBase::FactoryContextImplBase(
     Server::Instance& server, ProtobufMessage::ValidationVisitor& validation_visitor,
     Stats::ScopeSharedPtr scope, Stats::ScopeSharedPtr listener_scope,
-    ListenerInfoConstSharedPtr listener_info)
+    const Network::ListenerInfoConstSharedPtr& listener_info)
     : server_(server), validation_visitor_(validation_visitor), scope_(std::move(scope)),
-      listener_scope_(std::move(listener_scope)), listener_info_(std::move(listener_info)) {
-  if (listener_info_ == nullptr) {
-    listener_info_ = std::make_shared<ListenerInfoImpl>();
-  }
```

Comment:
nit: `ASSERT(listener_info_ == nullptr);` to avoid unexpected regression.

Reply:
Done.